alphastats/plots/VolcanoPlot.py

```diff
@@ -62,20 +76,39 @@ def __init__(
         self.pvalue_column = None
         self.perm = perm
         self.color_list = color_list
+
+        if isinstance(group1, list) and isinstance(group2, list):
+            self.metadata, self.column = self._add_metadata_column(
+                metadata, group1, group2
+            )
+            self.group1, self.group2 = "group1", "group2"
+        else:
+            self.metadata, self.column = metadata, column
+            self.group1, self.group2 = group1, group2
+
         self._check_input()
 
+        self._statistics = Statistics(
+            mat=self.mat,
+            metadata=self.metadata,
+            sample=self.sample,
+            index_column=self.index_column,
+            preprocessing_info=self.preprocessing_info,
+        )
+
         if plot:
             self._perform_differential_expression_analysis()
             self._annotate_result_df()
             self._add_hover_data_columns()
             self._plot()
 
-    # TODO this changes the actual metadata .. is this intended?
-    def _add_metadata_column(self, group1_list: list, group2_list: list):
+    # TODO this used to change the actual metadata .. was this intended?
```

Comment:
Do I get this right:
It used to add the _comparison_column to the DataSet metadata.
Now it returns the altered metadata to VolcanoPlot, which serves the purporse during the Volcano plot generation, but it is never back propagated to the metadata attribute of DataSet?
If that is so, then this is now fine.

Reply:
exactly. It overwrites the instance variable `self.metadata` with the updated metadata, but does not change the original value.

Note that these instance variables (`mat`, `metadata`, ...) in principle still can be changed in case they are mutable  `pd.DataFrame`, `dict`, ..:
if in `Statistics` something like `self.mat["new_column"] = 123" is done, then it will change the actual `mat` object of the `DataSet` (this fact is used to update `preprocessing_info`)